File: src/api/discussion-create-draft.ts
import { type Discussion, DiscussionCommand, type SessionHandle } from "~/models";
import { discussionPostCommand } from "./private/discussion-post-command";
import { createEntityID } from "./helpers/entity-id";
import { discussions } from "./discussions";
import { discussionMessages } from "./discussion-messages";

export const discussionCreateDraft = async (session: SessionHandle, discussion: Discussion, content: string, replyTo = discussion.messages?.defaultReplyMessageID): Promise<void> => {
  await discussionPostCommand(session, DiscussionCommand.brouillon, {
    id: createEntityID(),
    content,
    replyMessageID: replyTo
  });

  await discussions(session, discussion.cache);
  await discussionMessages(session, discussion);
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/discussion-create-draft.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.api.helpers.createEntityID
import ink.literate.pawnote.api.private.discussionPostCommand
import ink.literate.pawnote.models.Discussion
import ink.literate.pawnote.models.DiscussionCommand
import ink.literate.pawnote.models.SessionHandle
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put

suspend fun discussionCreateDraft (session: SessionHandle, discussion: Discussion, content: String, replyTo: String? = discussion.messages?.defaultReplyMessageID) {
    discussionPostCommand(session, DiscussionCommand.brouillon, buildJsonObject {
        put("id", createEntityID())
        put("content", content)
        put("replyMessageID", replyTo)
    })

    discussions(session, discussion.cache)
    discussionMessages(session, discussion)
}